docker/CMakeLists.txt

```diff
@@ -0,0 +1,77 @@
+set(PARFLOW_DOCKER_BASE_IMAGE "ubuntu:20.04" CACHE STRING "Docker base image to use")
+set(PARFLOW_DOCKER_VERSION "py-pftools" CACHE STRING "Version of ParFlow to install in docker (v3.6.0, master)")
```

Comment:
This was to enable some freedom on the configuration side. But we can remove them and assume an advanced user could run the docker command themselves while providing those additional informations. That will reduce the build complexity by just having sensible defaults. 

Moreover we could copy the Parflow source into the docker rather than cloning it via git. That way, no version to worry about, that will always be what you have locally. What do you think @smithsg84 ? Should I go that way?